File: app/src/main/java/com/teamunemployment/lolanalytics/front_page/Search/Model/ChampSearchCardView.java
package com.teamunemployment.lolanalytics.front_page.Search.Model;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.squareup.picasso.Picasso;
import com.teamunemployment.lolanalytics.data.model.Champ;
import com.teamunemployment.lolanalytics.front_page.Search.SearchPresenter;
import com.teamunemployment.lolanalytics.R;


import butterknife.BindView;
import butterknife.ButterKnife;
import com.mikhaellopez.circularimageview.CircularImageView;

/**
 * Created by Josiah Kendall
 */

public class ChampSearchCardView extends RecyclerView.ViewHolder implements View.OnClickListener {

    private SearchPresenter searchPresenter;
    private Champ champ;
    private Context context;

    @BindView(R.id.champ_circle_icon) CircularImageView champImage;

    public ChampSearchCardView(View itemView, SearchPresenter searchPresenter, Context context) {
        super(itemView);
        ButterKnife.bind(this, itemView);
        itemView.setOnClickListener(this);
        this.searchPresenter = searchPresenter;
        this.context = context;
    }

    /**
     * Set the champ that this card/item relates to.
     * @param champ
     */
    public void SetChamp(Champ champ) {
        this.champ = champ;
        if (champ.getChampUrl().isEmpty()) {
            // set our clear filter
            champImage.setImageResource(R.drawable.ic_account_circle_black_24dp);
            return;
        }
        Picasso.with(context).load(champ.getChampId()).into(champImage);
    }

    @Override
    public void onClick(View view) {
        //searchPresenter.
        searchPresenter.HandleChampClick(champ);
    }
}

File: app/src/main/java/com/teamunemployment/lolanalytics/front_page/Search/Model/ChampSearchCardView.kt
package com.teamunemployment.lolanalytics.front_page.Search.Model

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.View

import com.squareup.picasso.Picasso
import com.teamunemployment.lolanalytics.data.model.Champ
import com.teamunemployment.lolanalytics.front_page.Search.SearchPresenter
import com.teamunemployment.lolanalytics.R
import kotlinx.android.synthetic.main.champ_search_card.view.*

/**
 * Created by Josiah Kendall
 */

class ChampSearchCardView(private val view: View, private val searchPresenter: SearchPresenter, private val context: Context) : RecyclerView.ViewHolder(view), View.OnClickListener {
    private var champ: Champ? = null


    init {
        view.setOnClickListener(this)
    }

    /**
     * Set the champ that this card/item relates to.
     * @param champ
     */
    fun SetChamp(champ: Champ) {
        this.champ = champ
        if (champ.champUrl.isEmpty()) {
            // set our clear filter
            view.champImage.setImageResource(R.drawable.ic_account_circle_black_24dp)
            return
        }
        Picasso.with(context).load(champ.champId).into(view.champImage)
    }

    override fun onClick(view: View) {
        //searchPresenter.
        searchPresenter.handleChampClick(champ)
    }
}
